src/components/common/DatePickerInput/index.tsx

```diff
@@ -1,71 +1,57 @@
+import { useFormContext, Controller } from 'react-hook-form'
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
 import { DatePicker } from '@mui/x-date-pickers/DatePicker'
 import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'
 import TextField from '@mui/material/TextField'
-import type { DateValidationError } from '@mui/x-date-pickers/internals/hooks/validation/useDateValidation'
-
-export const _getDateError = (reason: DateValidationError) => {
-  switch (reason) {
-    case 'invalidDate': {
-      return 'Invalid date'
-    }
-    case 'shouldDisableDate': {
-      return 'Cannot select chosen date'
-    }
-    case 'disableFuture': {
-      return 'Date cannot be in the future'
-    }
-    case 'disablePast': {
-      return 'Date cannot be in the past'
-    }
-    // 'minDate'/'maxDate' are errors from date-fns
-    case 'minDate':
-    case 'maxDate':
-    default: {
-      return undefined
-    }
-  }
-}
+import isFuture from 'date-fns/isFuture'
+import isValid from 'date-fns/isValid'
 
 const DatePickerInput = ({
   name,
   label,
-  rules,
+  deps,
+  disableFuture = true,
+  validate,
 }: {
   name: string
   label: string
-  rules?: UseControllerProps['rules']
+  deps?: string[]
+  disableFuture?: boolean
+  validate?: (value: Date | null) => string | undefined
 }) => {
-  const { control, setError } = useFormContext()
+  const { control } = useFormContext()
 
   return (
     <Controller
       name={name}
       control={control}
-      rules={rules}
+      rules={{
+        deps,
+        validate: (val) => {
+          if (!val) {
+            return
+          }
+
+          if (!isValid(val)) {
+            return 'Invalid date'
+          }
+
+          if (disableFuture && isFuture(val)) {
+            return 'Date cannot be in the future'
+          }
+
+          return validate?.(val)
+        },
+      }}
       render={({ field, fieldState }) => (
         <LocalizationProvider dateAdapter={AdapterDateFns}>
           <DatePicker
             label={label}
+            inputFormat="dd/MM/yyyy"
             {...field}
-            disableFuture
-            onError={(reason) => {
-              const message = _getDateError(reason)
-              if (message) {
-                setError(field.name, {
-                  type: reason as string,
-                  message,
-                })
-              }
-            }}
-            renderInput={({ label, error, ...params }) => (
-              <TextField
-                label={fieldState.error?.message || label}
-                {...params}
-                fullWidth
-                error={!!fieldState.error || error}
-              />
+            disableFuture={disableFuture}
+            renderInput={({ label, error: _, ...params }) => (
+              <TextField label={fieldState.error?.message || label} {...params} fullWidth error={!!fieldState.error} />
```

Comment:
Is there a reason we are not using `error` as a fallback anymore? Now its possible to type a date like 01/07/0022 whereas before the validation would fail.

Reply:
The `<DatePicker>` has it's own `error` flag, unrelated to RHF. It was causing validation collisions so we now rely purely on the RHF `fieldState.error` instead.